packages/cli/test/unit/commands/db/revert.test.ts

```diff
@@ -74,6 +74,45 @@ test("don't revert the last migration if it had no down migration", async () =>
 	expect(dataSource.destroy).not.toHaveBeenCalled();
 });
 
+test('print migration name in error message is the migration has a name', async () => {
+	//
+	// ARRANGE
+	//
+	class TestMigration implements IrreversibleMigration {
+		name = 'Test Migration';
+
+		async up() {}
+	}
+
+	const connectionOptions = DbConfig.getConnectionOptions();
+	const migrations = [TestMigration];
+	// @ts-expect-error property is readonly
+	connectionOptions.migrations = migrations;
+	const dataSource = mock<DataSource>();
+	// @ts-expect-error property is readonly, and I can't pass them the `mock`
+	// because `mock` will mock the down method and thus defeat the purpose
+	// of this test, because the tested code will assume that the migration has a
+	// down method.
+	dataSource.migrations = migrations.map((M) => new M());
+
+	//
+	// ACT
+	//
+	await main(connectionOptions, logger, function () {
+		return dataSource;
+	} as never);
+
+	//
+	// ASSERT
+	//
+	expect(logger.error).toHaveBeenCalledTimes(1);
+	expect(logger.error).toHaveBeenCalledWith(
+		'The last migration "Test Migration" was irreversible and cannot be reverted.',
+	);
```

Comment:
Just as with error messages, should we avoid asserting on specific strings?

Reply:
If you ask me I'd say it depends.

If we just want to make sure that **some error** is thrown then we can apply DRY here similar to how it already is done with `UM_FIX_INSTRUCTION`.
On the other hand, matching against a literal makes sense if the error became part of the external contract and we want to have an extra assertion making sure it's not changed without acknowledging that the contract changed. E.g. if I change a constant and all tests pass I don't expect the FE to break. In that case asserting against a literal creates value because every time I change a test I have to assume that the FE could break.

In this case I can create a constant for the constant error message, for the dynamic one I will have to continue to construct it in the tests.